accepted/00000-auto-generate-bootstrap-repositories.md

```diff
@@ -0,0 +1,71 @@
+[motivation]: #motivation
+
+To be able to register a system to Uyuni Server you need to have the software installed which provide the required tooling.
+
+A bootstrap repository provide the software for the supported operating systems with its dependencies. The content for the
+bootstrap repositories come from the normal channels which must be synchronized and updated on regular base.
+
+Users very often do not know this, or forget to create them when they synchronize a new product. Also the bootstrap repository
+is only updated when the user execute the generator script again.
+
+This RFC should describe how to change the current tools to automatically 
+execute the generation of bootstrap repositories after
+a new product was added and after a channel changed when a maintenance update was synced to Uyuni Server.
+
+# Detailed design
+[design]: #detailed-design
+
+The currently exiting tool `mgr-create-bootstrap-repo` list all available bootstrap repositories it could generate.
+This list depends on the products the user has synchronized.
+
+Step 1: provide an option to enable an automatic mode which iterate over all available bootstrap repositories and generate them
+when needed. As we have sometimes have multiple options for the same local "path" we should iterate over the paths to minimize
+the number of runs.
+
+Step 2: add a log file to `mgr-create-bootstrap-repo`. Currently it has no log file. But when this tool is running in the background
+we need the possibility to debug problems. All messages should go to a log file.
+
+Step 3: let `mgr-create-bootstrap-repo` use a configuration file. All commandline options should be possible to set via
+configuration file. This allows to influence the behavior even in automatic mode when it is not possible to change the commandline
+options.
+
+To trigger the re-generation we implement the following algorithm:
+
+- Create a taskomatic job running every 15 minutes and call `mgr-create-bootstrap-repo --auto`
+- It calculate all channels which are needed to create the bootstrap repository for every available OS
+- It compare the `last_modified` timestamps of the channels (DB) with the `modified date` of the main metadata file
+  (repomd.xml or Release)
+- if all `last_modified` timestamps are newer then the timestamp of the main metadata file, re-generate the bootstrap repository
+- if any `last_modified` timestamps are newer then the timestamp of the main metadata file and the `modified date` of the main metadata
+  file is older than 26 hours, re-generate the bootstrap repository. This is for the case a channel is not synced. It can happen on
+  custom repositories needed for bootstrap repositories.
+
+Provide a `--force` option to enforce recreation of all bootstrap repositories without checking if needed. Mostly for manual usage.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+We maybe 15 minutes late with the bootstrap repository or maybe longer when a channel failed to sync or does not sync at all.
```

Comment:
Let's think that the user is enabling a SUSE product from WebUI, or adding something external manually or with `spacewalk-common-channels`.

How would the user know when it's possible to start bootstrapping stuff?

Reply:
> Let's think that the user is enabling a SUSE product from WebUI, or adding something external manually or with `spacewalk-common-channels`.
> 
> How would the user know when it's possible to start bootstrapping stuff?

You are right: we should provide feedback to the user that the bootstrap repository is ready. 

Do we provide any feedback now?

Would it make sense to consider a repository/channel is not fully synced until the bootstrap repository creation is not finished?

Reply:
This is not a new problem. It already exist. What I can add to the RFC is, that we can send a notification to users.
But I see this as optional and a nice to have enhancement.

Reply:
FMPOV, the problem is that until now they had to run it manually, but they knew that if the script didn't show errors, they could already bootstrap.

While now they won't need this, but they don't really know when they can start to bootstrap.

So yeah, maybe having a notification is optional and a "nice to have", 

I am OK with but if we do it this way, we need to clearly document at the docs that they should track the logs.

Reply:
I added a section about Notifications to the text